elephant/kernels.py

```diff
@@ -0,0 +1,339 @@
+            a time interval.
+        :type t: Quantity 1D
+        :returns: The result of the kernel evaluations.
+        :rtype: Quantity 1D
+        """
+        if not (isinstance(t, pq.Quantity)):
+            raise TypeError("The argument of the kernel callable must be "
+                            "of type quantity!")
+
+        if t.dimensionality.simplified != self.sigma.dimensionality.simplified:
+            raise TypeError("The dimensionality of sigma and the input array "
+                            "to the callable kernel object must be the same. "
+                            "Otherwise a normalization to 1 of the kernel "
+                            "cannot be performed.")
+
+        self._sigma_scaled = self.sigma.rescale(t.units)
+        # A hidden variable _sigma_scaled is introduced here in order to avoid
+        # accumulation of floating point errors of sigma upon multiple
+        # usages of the __call__ - function for the same Kernel instance.
+
+        return self._evaluate(t)
+
+    def _evaluate(self, t):
+        """ Evaluates the kernel.
+
+        :param t: Interval on which the kernel is evaluated, not necessarily
+            a time interval.
+        :type t: Quantity 1D
+        :returns: The result of the kernel evaluations.
+        :rtype: Quantity 1D
+        """
+        raise NotImplementedError()
+
+    def boundary_enclosing_at_least(self,
+                                    fraction=default_kernel_area_fraction):
+        """ Calculates the boundary :math:`b` so that the integral from
+        :math:`-b` to :math:`b` encloses at least a certain fraction of the
+        integral over the complete kernel.
+
+        :param float fraction: Fraction of the whole area which at least has
+            to be enclosed.
+        :returns: boundary
+        :rtype: Quantity scalar
+        """
+        raise NotImplementedError()
+
+    def m_idx(self, t):
+        """
+        Estimates the index of the Median of the kernel.
+        This parameter is not mandatory for symmetrical kernels but it is
+        required when asymmetrical kernels have to be aligned at their median.
+
+        :param t: Interval on which the kernel is evaluated,
+        :type t: Quantity 1D
+        :returns: Estimated value of the kernel median
+        :rtype: int
+
+        Remarks:
+        The formula in this method using retrieval of the sampling interval
+        from t only works for t with equidistant intervals!
+        The formula calculates the Median slightly wrong by the potentially
+        ignored probability in the distribution corresponding to lower values
+        than the minimum in the array t.
+        """
+        return np.nonzero(self(t).cumsum() *
+                          (t[len(t)-1] - t[0]) / (len(t) - 1) >= 0.5)[0].min()
+
+    def is_symmetric(self):
+        """ Should return `True` if the kernel is symmetric. """
+        return False
```

Comment:
I think returning 'False' is correct here. In the case of symmetric kernels, this method is overwritten in the class SymmetricKernel and then returns 'True', remaining with 'False' for the asymmetric ones.